src/rules/declaration-empty-line-before/index.js

```diff
@@ -0,0 +1,101 @@
+import {
+  blockString,
+  isCustomProperty,
+  isSingleLineString,
+  isStandardSyntaxDeclaration,
+  isStandardSyntaxProperty,
+  optionsHaveException,
+  optionsHaveIgnored,
+  report,
+  ruleMessages,
+  validateOptions,
+} from "../../utils"
+
+export const ruleName = "declaration-empty-line-before"
+
+export const messages = ruleMessages(ruleName, {
+  expected: "Expected empty line before declaration",
+  rejected: "Unexpected empty line before declaration",
+})
+
+export default function (expectation, options) {
+  return (root, result) => {
+    const validOptions = validateOptions(result, ruleName, {
+      actual: expectation,
+      possible: [
+        "always",
+        "never",
+      ],
+    }, {
+      actual: options,
+      possible: {
+        except: [
+          "first-nested",
+          "after-comment",
+          "after-declaration",
+        ],
+        ignore: [
+          "inside-single-line-block",
+        ],
+      },
+      optional: true,
+    })
+    if (!validOptions) { return }
+
+    root.walkDecls(decl => {
+      const { prop, parent } = decl
+
+      if (!isStandardSyntaxDeclaration(decl)) { return }
+      if (!isStandardSyntaxProperty(prop)) { return }
```

Comment:
I'm in favour of pushing anything like that to a plugin. Even if it means duplicating logic within it. *Consistently* distancing, by using the `isStandardSyntax*` utils, the core rules from non-standard syntax feels more sustainable and feels like we're keeping that can of worms securely sealed.